### ISSUE TITLE
PDF/UA page header and footer

### ISSUE THREAD

**Issue by jjsarton** [Kozea/WeasyPrint#1836]
Page header and footer content are not marked as artifact and therefor spoken from a screen reader. This is wrong!

**Comment by liZe** [comment]
Hi!

Could you please share an example, so that we can perfectly understand what you are talking about?

**Comment by liZe** [comment]
If you’re interested in this issue, it’s time to test https://github.com/Kozea/WeasyPrint/pull/2471! Feedback will be highly appreciated, even to say that it just works. 🙏

**Comment by oraslaci** [comment]
Hi!
Really awesome progress on PDF/UA, but unfortunately this issue still persists.
Acrobat reader reads aloud the contents of the footer first thing on every page.
@jjsarton is right, according to recommendations. headers and footers should be tagged as artifiacts, so they are avoided by screenreaders.
Have a good one,
Laszlo

**Comment by liZe** [comment]
> according to recommendations. headers and footers should be tagged as artifiacts

That’s supposed to be the case, but it looks like it’s not. I’ll take a look at that.

**Comment by liZe** [comment]
That’s a bit complex.

What we currently do is to mark the content of page margins, but we don’t include the marked content in the main content tree, so that it doesn’t appear in PAC for example.

What we’d like to do is to include the marked content of margins in Artifact nodes at the root of the main content tree. But this is only possible in PDF 2.0 (PDF/UA-2), not before.

What the PDF Association asks is to mark the content of page margins as Artifacts. It means that we can’t tag the inner content. Currently, there’s no way to do this easily in WeasyPrint.

@oraslaci In the current PDFs, is margin content really read by screenreaders? It’s present in the list of tagged content by Adobe (because the content is actually tagged), but if screenreaders use the main content tree (they probably do, to get the right order?), we could probably keep this feature for later.

**Comment by nathalie-tate** [comment]
Hello, I just wanted to note that in my testing, the NVDA screenreader does in fact read page footers aloud on every page 

eta: it doesn't *always* do it. I suspect it may have something to do with the selected settings in adobe acrobat. sorry for any confusion.

**Comment by liZe** [comment]
> Hello, I just wanted to note that in my testing, the NVDA screenreader does in fact read page footers aloud on every page
> 
> eta: it doesn't _always_ do it. I suspect it may have something to do with the selected settings in adobe acrobat. sorry for any confusion.

Hi @nathalie-tate 

Thanks for your report, that’s really useful to get feedback from people using screenreaders 🙏 

Do you have an idea of which settings are useful in Acrobat, so that we could learn more about how we could improve the rendering? It could also depend on the way data are put in headers/footers, so if you have different documents generated by WeasyPrint that give different results, don’t hesitate to share them!

**Comment by nathalie-tate** [comment]
so, I *thought* it might be related to the Reading>Reading Order Options settings, but now I can't replicate the issue; I can't get nvda to read the footer at all. I'm not terribly good at using nvda, so I may have been doing something wrong before. 

I'm also starting to suspect that there might actually be something wrong with my install of acrobat, because the built in read out loud feature seems to randomly stop working. so my testing might be of questionable utility. 

if it's helpful at all, this is the document I'm testing with: https://legislature.mi.gov/documents/mcl/pdf/MCL-ACT-7-OF-1955.pdf 

the footer is populated using the @page, @bottom-left, @bottom-center, and @bottom-right css rules 

again, sorry for any confusion
